proto/directions.proto

```diff
@@ -0,0 +1,94 @@
+package valhalla;
+
+message Waypoint {
+  optional float lat = 1;
+  optional float lon = 2;
+  optional string type = 3;
+  optional uint32 bearing = 4;
+  optional string name = 5;
+  optional string street = 6;
+  optional string side_of_street = 7;
+}
+
+message BoundingBox {
+  optional float min_lat = 1;
+  optional float min_lon = 2;
+  optional float max_lat = 3;
+  optional float max_lon = 4;
+}
+
+message StreetName {
+  optional string text = 1;  // Text of street name, examples: MD 43 East, White Marsh Boulevard 
+  optional bool is_ref = 2;  // true if street name is a ref, examples: true, false
+}
+
+message Step {
+  optional float distance = 1;
+  optional uint32 duration = 2;
+  optional uint32 geometry_index_begin = 3;      // begin geometry inclusive index
+  optional uint32 geometry_index_end = 4;        // end geometry inclusive index
+  optional uint32 maneuver_index_begin = 5;      // the index of the maneuver at the start of the step
```

Comment:
what about `before_maneuver_index` and `after_maneuver_index`? I feel like begin and end make it seem like its a range.

Reply:
`maneuver_into_step` might be a better name to indicate the directionality of the maneuver.

Reply:
what about `incoming_maneuver_index` and `outgoing_maneuver_index` ?

Reply:
yeah my worry with having it the same as the geometry naming was that the geometry naming was actually a range - like from 0 to 10, whereas this is two distinct numbers. 

I like `incoming_maneuver_index` and `outgoing_maneuver_index`, that works for me